gpbft/gpbft.go

```diff
@@ -981,18 +981,21 @@ func (q *quorumState) ListStrongQuorumValues() []ECChain {
 	var withQuorum []ECChain
 	for cid, cp := range q.chainSupport {
 		if cp.hasStrongQuorum {
-			withQuorum = append(withQuorum, q.chainSupport[cid].chain)
+			withQuorum = append(withQuorum, filterEmptyTipsets(q.chainSupport[cid].chain))
 		}
 	}
 	sort.Slice(withQuorum, func(i, j int) bool {
-		return len(withQuorum[i]) > len(withQuorum[j])
+		return withQuorum[i].Head().Epoch > withQuorum[j].Head().Epoch
 	})
-	prevLength := 0
-	for _, v := range withQuorum {
-		if len(v) == prevLength {
-			panic(fmt.Sprintf("multiple chains of length %d with strong quorum", prevLength))
+	if len(withQuorum) <= 1 {
+		return withQuorum
+	}
+
+	// Check that the chains are prefixes of each other.
```

Comment:
This is a different check than the existing one. A direct translation of the existing one would just replace `len(v)` with `v.Head().Epoch`, and I think we should have that check.

I can't work out what the new check is doing, or if we need it. Can you explain it more, and why just checking the Epoch isn't enough?

Reply:
Since chains are sorted now by decreasing head epoch number, we can just start with the first and make sure one is a prefix of each other. The previous test was wrong as you noted due to the possibility of null epochs, and just checking the previous head epoch with the subsequent head epoch would not work (which is what I understand by you saying "just replace `len(v)` with `v.Head().Epoch`). Or put differently, what I do here is what I understand as the most natural way of just replacing `len(v)` with `v.Head().Epoch`.